low-level/actuators/impl/generic/realstor_encl.py

```diff
@@ -612,3 +619,60 @@ def _get_encl_response(self, uri, request_type):
         except ValueError as err:
             logger.error(f"invalid data for {uri} {err}")
             return None
+
+    def _put_enclosure_action(self, resource, enclosure_request):
+        len_resource = len(resource.split())
+        invalid_args = False
+        ctrl_cmd = resource
+        ctrl_status = ''
+        if resource.startswith('shutdown') and len_resource <= 2:
+            ctrl_status = 'shutdown'
+            if len_resource == 1:
+                ctrl_cmd = f'{resource} both'
+        elif resource.startswith('restart sc') and len_resource <= 3:
+            ctrl_status = 'restart sc'
+            if len_resource == 2:
+                ctrl_cmd = f'{resource} both'
+        elif resource.startswith('restart mc') and len_resource <= 3:
+            ctrl_status = 'restart mc'
+            if len_resource == 2:
+                ctrl_cmd = f'{resource} both'
+        else:
+            invalid_args = True
+
+        if not ctrl_cmd.endswith((' a', ' b', ' both')):
+            invalid_args = True
+
+        if invalid_args:
+            err_msg = "Invalid resource {} for an {} actuator request".format(
+                resource, enclosure_request)

```

Comment:
plz add sample log statement here, need to see the variable values

Reply:
Done